eZ/Publish/Core/Limitation/LocationLimitationType.php

```diff
@@ -131,12 +134,16 @@ public function evaluate(APILimitationValue $value, APIUserReference $currentUse
             );
         }
 
+        $targets = array_filter($targets, function ($target) {
+            return !$target instanceof Version;
```

Comment:
The actual bug exists in the implementation of `\eZ\Publish\Core\Repository\Permission\PermissionResolver::lookupLimitations`. 
An instance of `Target` should be passed to `TargetAwareType` Limitation type only, which [`canUser` takes into the account](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.4/eZ/Publish/Core/Repository/Permission/PermissionResolver.php#L224). The `lookupLimitation` method needs to behave the same way. // cc @mikadamczyk 

While this fix works for `LocationLimitation`, there might be similar issues 

Ref:
- `\eZ\Publish\SPI\Limitation\TargetAwareType`,
- `\eZ\Publish\SPI\Limitation\Target`.